File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        def dfs(start, currentXor):
            if start == len(nums):
                nonlocal sum
                sum += currentXor
                return

            dfs(start+1, currentXor)
            dfs(start+1, currentXor ^ nums[start])

        sum = 0
        dfs(0, 0)

        return sum

File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var subsetXORSum = function (nums) {
    const dfs = function (start, currentXor) {
        if (start === nums.length) {
            sum += currentXor;
            return;
        }

        dfs(start + 1, currentXor);
        dfs(start + 1, currentXor ^ nums[start]);
    }

    let sum = 0;
    dfs(0, 0);

    return sum;
};